src/openforms/translations/tests/test_api.py

```diff
@@ -0,0 +1,131 @@
+            self.assertEqual(available, expected)
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+    )
+    def test_info_contains_active_language(self):
+        # default fallback
+        url = reverse("api:i18n:info")
+        response = self.client.get(url)
+        current = response.json()["current"]
+        self.assertEqual(current, "en")
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+        MIDDLEWARE=[
+            "django.contrib.sessions.middleware.SessionMiddleware",
+            "django.middleware.locale.LocaleMiddleware",
+            "django.middleware.common.CommonMiddleware",
+        ],
+    )
+    def test_info_negotiates_active_language_from_header(self):
+        url = reverse("api:i18n:info")
+        response = self.client.get(
+            url, HTTP_ACCEPT_LANGUAGE="nl-BE, nl;q=0.9, en;q=0.8"
+        )
+        current = response.json()["current"]
+        self.assertEqual(current, "nl")
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+        LANGUAGE_COOKIE_NAME="openforms-language",
+        MIDDLEWARE=[
+            "django.contrib.sessions.middleware.SessionMiddleware",
+            "django.middleware.locale.LocaleMiddleware",
+            "django.middleware.common.CommonMiddleware",
+        ],
+    )
+    def test_put_overrides_negotiated_current(self):
+        SELECT = "en"
+        url = reverse("api:i18n:language")
+        response = self.client.put(
+            url,
+            data={"code": SELECT},
+            HTTP_ACCEPT_LANGUAGE="nl-BE, nl;q=0.9, en;q=0.8",
+        )
+        self.assertEqual(response.status_code, 204)
+
+        url = reverse("api:i18n:info")
+        response = self.client.get(
+            url,
+            HTTP_ACCEPT_LANGUAGE="nl-BE, nl;q=0.9, en;q=0.8",
+        )
+        current = response.json()["current"]
+        self.assertEqual(current, SELECT)
+
+    @override_settings(
+        USE_I18N=True,
+        LANGUAGE_CODE="en",
+        LANGUAGES=[
+            ("en", _("English")),
+            ("nl", _("Dutch")),
+        ],
+        LANGUAGE_COOKIE_NAME="openforms-language",
+        MIDDLEWARE=[
+            "django.contrib.sessions.middleware.SessionMiddleware",
+            "django.middleware.locale.LocaleMiddleware",
+            "django.middleware.common.CommonMiddleware",
+        ],
+    )
+    def test_put_language_checks_for_availability(self):
+        url = reverse("api:i18n:language")
+        # ትግርኛ is not available
+        response = self.client.put(url, data={"code": "ti"})
+        self.assertEqual(response.status_code, 400)
```

Comment:
you'll also want to make an assertion that the `name` and `code` of `invalidParams[0]` is as expected